t/api/04-jobs.t

```diff
@@ -523,21 +524,45 @@ subtest 'update job and job settings' => sub {
     $t->get_ok('/api/v1/jobs/99926')->status_is(200);
     $t->json_is('/job/group' => undef, 'group removed');
 
+
     # set machine
     $t->put_ok(
         '/api/v1/jobs/99926',
         json => {
             settings => {
-                MACHINE => '64bit'
+                MACHINE      => '64bit',
+                WORKER_CLASS => ':UFP:NCC1701F',
             }})->status_is(200);
     $t->get_ok('/api/v1/jobs/99926')->status_is(200);
     $t->json_is(
         '/job/settings' => {
-            NAME    => '00099926-@64bit',
-            MACHINE => '64bit',
+            NAME         => '00099926-@64bit',
+            MACHINE      => '64bit',
+            WORKER_CLASS => ':UFP:NCC1701F',
         },
-        'also name update, all other settings cleaned'
+        'also name and worker class updated, all other settings cleaned'
     );
 };
 
+subtest 'filter by worker_class' => sub {
+
+    $query->query(worker_class => ':MiB:');
+    $get = $t->get_ok($query->path_query)->status_is(200);
+    $res = $get->tx->res->json;
+    ok(!@{$res->{jobs}}, 'Worker class does not exist');
+
+    $query->query(worker_class => '::');
```

Comment:
but what happens if you query :": ? i.e. you can't generate SQL injections with that, right? Passing parameters to like sounds dangerous in general.

Reply:
Yeah, I think so too, at one point I thought about doing it /jobs/proxy/${word} but looked to me the same. However the `/^:\w+:/` in the regexp guards against strage stuff between the colons, and since we're using DBIx's internal search.

Reply:
true that